snapcraft/cli/__init__.py

```diff
@@ -89,6 +89,7 @@ def get_command(self, ctx, cmd_name):
 @click.option('--debug', '-d', is_flag=True)
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
+    ctx.obj['debug'] = debug
     if debug:
```

Comment:
But you removed the use of obj in get_project_options... I thought it works like other options (geoip, parallel_builds, target_arch) now via params?